### PR TITLE
Implementation of /reddit endpoint closes #28

### LINKED ISSUE #28: Map reddit api data to meet interface of integration module
**User Story**
As a backend developer, I'd like to map the Reddit API data such that it meets the interface defined by the integration module so that it can be more easily understood when combining data from multiple social media APIs.

**Acceptance Criteria**
- Define fields of a `RedditData` type, following the [guidelines](https://github.com/softeng-701-group-5/softeng-701-assignment-1/wiki/Frontend-Meeting) defined by the front-end 
- All current data types fetched from the Reddit API are mapped to the `RedditData` type
- Builder should be used to create a `RedditData` type, similar to how `DemoData` is done
- Implement `getFeed()` to return a `Flux<RedditData>` type, (interface defined in #43)
- Unit tests are added and/or updated

**Notes**

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**

To connect the reddit API data to a server endpoint

**Describe the solution you'd like**

Implement the `FeedProvider<RedditData>` interface defined in #43 

**Describe alternatives you've considered**

The original approach was considered, but compared to the interface approach it is not simple to integrate with other feeds.

### PR DESCRIPTION
**Pull request by stak469**
GitHub Issue (If applicable): #28

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

The Reddit endpoint (/reddit) currently does not do anything.

## What is the new behavior?

The Reddit endpoint correctly returns the JSON data of all the submissions on the users front page.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [x] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

The credentials for application.properties are on discord.
Internal Issue (If applicable):

### CODE REVIEW

File: server/src/main/java/com/feeder/server/model/RedditData.java [hunk 1]
```diff
@@ -1,3 +1,79 @@
...
+  public abstract Date created();
+
+  @JsonProperty("subreddit")
+  public abstract String subreddit();
+
+  @JsonProperty("url")
+  public abstract String url();
+
+
+  /**
```
Comment (PreetPatel) [inline server/src/main/java/com/feeder/server/model/RedditData.java]: Please remove these comments

Comment (jordansimsmith) [reply server/src/main/java/com/feeder/server/model/RedditData.java]: I think the comment is valid. It explains why 'self' is returned (not obvious behaviour)

Comment (PreetPatel) [reply server/src/main/java/com/feeder/server/model/RedditData.java]: No i didnt comment on that.. i commented on the comment about optional usage.

**Comment by jordansimsmith** [comment]
Tested locally with new upstream changes, still works well :tada:

**Pull request merged**